plainly-plugin/src/ui/hooks/useSettings.ts

```diff
@@ -86,8 +97,41 @@ export const useSettings = () => {
     [],
   );
 
-  // TODO: handle decrypt
-  const getSettingsApiKey = () => settings.apiKey;
+  const getSettingsApiKey = useCallback(
```

Comment:
imo this is not good, here's why:

* the return should be the key
* the error should be thrown
* the logic must be:
  * pin given, not encryped -> error
  * pin not given, encryped -> error
  * pin given, encrypted -> decrypt
  * pint not given, not encryped -> plain text